File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/extractors/SimplePageImgProcessor.java
package ru.kmorozov.gbd.core.logic.extractors;

import ru.kmorozov.gbd.core.config.IStoredItem;
import ru.kmorozov.gbd.core.logic.Proxy.HttpHostExt;
import ru.kmorozov.gbd.core.logic.context.BookContext;
import ru.kmorozov.gbd.core.logic.extractors.base.AbstractPageImgProcessor;
import ru.kmorozov.gbd.core.logic.model.book.base.AbstractPage;

/**
 * Created by sbt-morozov-kv on 18.11.2016.
 */
public class SimplePageImgProcessor<T extends AbstractPage> extends AbstractPageImgProcessor {

    public SimplePageImgProcessor(BookContext bookContext, T page, HttpHostExt usedProxy) {
        super(bookContext, page, usedProxy);
    }

    @Override
    protected String getErrorMsg(String imgUrl, HttpHostExt proxy) {
        return String.format("Cannot load data from %s", imgUrl);
    }

    @Override
    protected String getSuccessMsg() {
        return String.format("Finished img processing for %s", this.page.getPid());
    }

    @Override
    public void run() {
        if (this.page.isDataProcessed()) return;

        this.processImage(this.page.getImgUrl());
    }

    @Override
    protected boolean validateOutput(IStoredItem storedItem, int width) {
        return true;
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/extractors/SimplePageImgProcessor.kt
package ru.kmorozov.gbd.core.logic.extractors

import ru.kmorozov.gbd.core.logic.proxy.HttpHostExt
import ru.kmorozov.gbd.core.logic.context.BookContext
import ru.kmorozov.gbd.core.logic.extractors.base.AbstractPageImgProcessor
import ru.kmorozov.gbd.core.logic.extractors.base.IUniqueRunnable
import ru.kmorozov.gbd.core.logic.model.book.base.AbstractPage

/**
 * Created by sbt-morozov-kv on 18.11.2016.
 */
open class SimplePageImgProcessor<T : AbstractPage>(bookContext: BookContext, page: T, usedProxy: HttpHostExt) :
    AbstractPageImgProcessor<T>(bookContext, page, usedProxy) {

    override val successMsg: String
        get() = "Finished img processing for ${uniqueObject.pid}"

    override fun getErrorMsg(imgUrl: String, proxy: HttpHostExt): String {
        return "Cannot load data from $imgUrl"
    }

    override fun run() {
        if (uniqueObject.isDataProcessed) return

        processImage(uniqueObject.imgUrl)
    }
}
